core/error_macros.h

```diff
@@ -115,28 +115,51 @@ extern bool _err_error_exists;
 #define FUNCTION_STR __FUNCTION__
 #endif
 
+// Don't use this directly; instead, use any of the CRASH_* macros
+#ifdef _MSC_VER
+#define GENERATE_TRAP                       \
+	__debugbreak();                         \
+	/* Avoid warning about control paths */ \
+	for (;;) {                              \
+	}
+#else
+#define GENERATE_TRAP __builtin_trap();
+#endif
+
 #define ERR_FAIL_INDEX(m_index, m_size)                                                                                    \
-	do {                                                                                                                   \
+	{                                                                                                                      \
 		if ((m_index) < 0 || (m_index) >= (m_size)) {                                                                      \
 			_err_print_error(FUNCTION_STR, __FILE__, __LINE__, "Index " _STR(m_index) " out of size (" _STR(m_size) ")."); \
 			return;                                                                                                        \
 		} else                                                                                                             \
 			_err_error_exists = false;                                                                                     \
-	} while (0);
+	}
 
 /** An index has failed if m_index<0 or m_index >=m_size, the function exists.
-  * This function returns an error value, if returning Error, please select the most
-  * appropriate error condition from error_macros.h
-  */
+* This function returns an error value, if returning Error, please select the most
+* appropriate error condition from error_macros.h
+*/
 
 #define ERR_FAIL_INDEX_V(m_index, m_size, m_retval)                                                                        \
-	do {                                                                                                                   \
```

Comment:
I now understand the benefits of the loop construct. And we can't refactor the macro into a function because we need the preprocessor magic of converting the checked expression into a string. So I will put it back. But is this the only macro needing it?
Regarding the trailing semicolon, maybe that is due to clang-format?